### DISCUSSION TITLE
Error: Cannot read properties of null (reading 'api_key')

### DISCUSSION THREAD

**Discussion by ppokrovskii** [superagent-ai/superagent#433] [Q&A] [answered]
Can someone please help me resolve the error below? I'm running frontent locally, backend in docker, supabase cloud platform.
In ui\.env I've setup NEXT_PUBLIC_SUPABASE_URL and NEXT_PUBLIC_SUPABASE_ANON_KEY 
but after login via GitHub account I'm getting redircted to /agents page and getting the error below.
It seems to be an issue with supabase api_key but I can't figure out where else to set it up?

> Unhandled Runtime Error
> Error: Cannot read properties of null (reading 'api_key')
> 
> Source
> app\agents\page.tsx (21:30) @ api_key

**Comment by homanp** [comment, marked as answer]
Hey @ppokrovskii this means that the backend isn't running or wasn't running when you signed up the user via UI. Delete the user and recreate with the backend running.

**Reply by ppokrovskii** [reply]
thanks! the issue was that I missed the step of "Supabase setup for Superagent UI project" and function that inserts a row into public.profiles
wasn't there on supabase side.
So user was not created completely.

Solution was to complete Supabase setup for Superagent UI project, delete user in supabase, clear cookies and recreate user from scratch

**Reply by homanp** [reply]
> thanks! the issue was that I missed the step of "Supabase setup for Superagent UI project" and function that inserts a row into public.profiles
> 
> wasn't there on supabase side.
> 
> So user was not created completely.
> 
> 
> 
> Solution was to complete Supabase setup for Superagent UI project, delete user in supabase, clear cookies and recreate user from scratch

Perfect! Ping if anything!
